python/tvm/relay/frontend/tflite.py

```diff
@@ -3588,8 +3643,14 @@ def from_tflite(model, shape_dict, dtype_dict):
     exp_tab = ExprTable()
     for model_input in model_inputs:
         model_input_name = get_tensor_name(subgraph, model_input)
-        shape = shape_dict[model_input_name] if model_input_name in shape_dict else None
-        dtype = dtype_dict[model_input_name] if model_input_name in dtype_dict else "float32"
+        if shape_dict:
+            shape = shape_dict[model_input_name] if model_input_name in shape_dict else None
+        else:
+            shape = get_tensor_shape(subgraph, model_input)
+        if dtype_dict:
+            dtype = dtype_dict[model_input_name] if model_input_name in dtype_dict else "float32"
+        else:
+            dtype = get_tensor_type(subgraph, model_input)
```

Comment:
We have a similar function, that collect the same information being proposed here in TVMC. I agree we should move what is in there, to unify functionality here.

Can you have a look on the function I'm pointing here (below) and spot why are they so different, and in case you agree on what's the best approach, improve it here and remove it there?

https://github.com/apache/tvm/blob/2999d03284c74f6840503ae3b880d3579a76f1af/python/tvm/driver/tvmc/frontends.py#L255-L278

Reply:
> We have a similar function, that collect the same information being proposed here in TVMC. I agree we should move what is in there, to unify functionality here.

Oh, it was there all along. I was loading my models in a separate script to put the relay output into my compile passes.


> Can you have a look on the function I'm pointing here (below) and spot why are they so different, 

I don't see much difference except that your code accounts for models with more than one subgraph.

> and in case you agree on what's the best approach, improve it here and remove it there?

My rationale behind making and putting this code in the tflite.py file was:
1. use the data in the graph since it is already embedded in it.
2. place the code inside the frontend code since it is dependent on the frontend.

Reply:
Cool. I think we both agree that it is better to have the funcionality only in the tflite.py, and remove it from TVMC.

So I suggest we keep the one that accounts for many subgraphs, and move it from TVMC to the official frontend? If you agree, feel free to do it in this PR.

Reply:
Thanks, I will update the PR.